src/MOI_wrapper.jl

```diff
@@ -2533,17 +2533,34 @@ function _has_primal_ray(model::Optimizer)
     return ret == 0
 end
 
-function MOI.get(model::Optimizer, attr::MOI.VariablePrimal, x::MOI.VariableIndex)
-    _throw_if_optimize_in_progress(model, attr)
-    MOI.check_result_index_bounds(model, attr)
-    key = model.has_unbounded_ray ? "UnbdRay" : "X"
+function _get_dbl_attr_variable(
+    model::Optimizer, key::String, x::MOI.VariableIndex
+)
     col = Cint(column(model, x) - 1)
     valueP = Ref{Cdouble}()
     ret = GRBgetdblattrelement(model, key, col, valueP)
     _check_ret(model, ret)
     return valueP[]
 end
 
+function MOI.get(
+    model::Optimizer, attr::MOI.VariablePrimal, x::MOI.VariableIndex
+)
+    _throw_if_optimize_in_progress(model, attr)
+    if model.has_unbounded_ray
+        return _get_dbl_attr_variable(model, "UnbdRay", x)
```

Comment:
We should only do that if `attr.N` is 1 I suppose. and we should also do `MOI.check_result_index_bounds` before that